x-pack/plugins/spaces/public/spaces_context/wrapper_internal.tsx

```diff
@@ -9,38 +9,38 @@ import type { PropsWithChildren } from 'react';
 import React, { useEffect, useMemo, useState } from 'react';
 
 import type { ApplicationStart, DocLinksStart, NotificationsStart } from 'src/core/public';
-import type { SpacesContextProps } from 'src/plugins/spaces_oss/public';
 
+import type { GetAllSpacesPurpose } from '../../common';
 import type { SpacesManager } from '../spaces_manager';
-import type { ShareToSpacesData, ShareToSpaceTarget } from '../types';
+import type { SpacesData, SpacesDataEntry } from '../types';
 import { createSpacesReactContext } from './context';
-import type { InternalProps, SpacesReactContext } from './types';
+import type { InternalProps, SpacesContextProps, SpacesReactContext } from './types';
 
 interface Services {
   application: ApplicationStart;
   docLinks: DocLinksStart;
   notifications: NotificationsStart;
 }
 
-async function getShareToSpacesData(
-  spacesManager: SpacesManager,
-  feature?: string
-): Promise<ShareToSpacesData> {
+async function getSpacesData(spacesManager: SpacesManager, feature?: string): Promise<SpacesData> {
   const spaces = await spacesManager.getSpaces({ includeAuthorizedPurposes: true });
   const activeSpace = await spacesManager.getActiveSpace();
   const spacesMap = spaces
-    .map<ShareToSpaceTarget>(({ authorizedPurposes, disabledFeatures, ...space }) => {
+    .map<SpacesDataEntry>(({ authorizedPurposes, disabledFeatures, ...space }) => {
       const isActiveSpace = space.id === activeSpace.id;
-      const cannotShareToSpace = authorizedPurposes?.shareSavedObjectsIntoSpace === false;
       const isFeatureDisabled = feature !== undefined && disabledFeatures.includes(feature);
       return {
         ...space,
         ...(isActiveSpace && { isActiveSpace }),
-        ...(cannotShareToSpace && { cannotShareToSpace }),
         ...(isFeatureDisabled && { isFeatureDisabled }),
+        isAuthorizedForPurpose: (purpose: GetAllSpacesPurpose) =>
+          // If authorizedPurposes is not present, then Security is disabled; normally in a situation like this we would "fail-secure", but
+          // in this case we are dealing with an abstraction over the client-side UI capabilities. There is no chance for privilege
+          // escalation here, and the correct behavior is that if Security is disabled, the user is implicitly authorized to do everything.
+          authorizedPurposes ? authorizedPurposes[purpose] === true : true,
       };
     })
-    .reduce((acc, cur) => acc.set(cur.id, cur), new Map<string, ShareToSpaceTarget>());
+    .reduce((acc, cur) => acc.set(cur.id, cur), new Map<string, SpacesDataEntry>());
```

Comment:
This function is run when the SpacesContext provider is first mounted. It fetches all spaces and passes them to consumers via the context `value`.

This was originally purpose-built for the share-to-space flyout, but I renamed it and added a generic `isAuthorizedForPurpose` function. I have two motivations behind this:
1. The CTS flyout can also reuse this data, so we don't need to fetch all spaces every time the flyout is rendered
2. Other consumers can reuse this data in the future so they don't have to fetch all spaces too